test/e2e/azure_test.go

```diff
@@ -91,6 +105,178 @@ var _ = Describe("Workoad cluster creation", func() {
 		})
 	})
 
+	Context("Create single cluster to validate the network policies", func() {
```

Comment:
If I'm reading this correctly, it creates a 3rd cluster as part of E2E? Should we merge this with the above one instead instead to speed up PR tests? The above test just creates the cluster without running any validation on it so I don't think it brings distinct coverage from creating the cluster and validating network policy.

Reply:
similar to https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/740/files#diff-8f80686adb74f19133b8b1728fbed453R126

@mboersma FYI since your PR overlaps with this one

Reply:
good point, +1 to Cecile's comment

Reply:
+1

Reply:
I will wait the comment in this PR https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/740/files#r448156634 to change things here